packages/react/src/transactions/useFollow.ts

```diff
@@ -24,82 +24,69 @@ import {
 
 import { Session, SessionType, useSession } from '../authentication';
 import { UseDeferredTask, useDeferredTask } from '../helpers/tasks';
-import { useLazyModuleMetadata } from '../misc';
 import { AsyncTransactionResult } from './adapters/AsyncTransactionResult';
 import { useFollowController } from './adapters/useFollowController';
 
 export class PrematureFollowError extends Error {
   name = 'PrematureFollowError' as const;
 }
 
-function useFollowRequestFactory() {
-  const { execute: fetchModule } = useLazyModuleMetadata();
-
-  return async (args: FollowArgs, session?: Session): Promise<FollowRequest> => {
-    invariant(
-      session?.authenticated,
-      'You must be authenticated to use this operation. Use `useLogin` hook to authenticate.',
-    );
-    invariant(
-      session.type === SessionType.WithProfile,
-      'You must have a profile to use this operation.',
-    );
-
-    const followPolicy = resolveFollowPolicy(args.profile);
-
-    switch (followPolicy.type) {
-      case FollowPolicyType.CHARGE:
-        return {
-          kind: TransactionKind.FOLLOW_PROFILE,
-          fee: {
-            amount: followPolicy.amount,
-            contractAddress: followPolicy.contractAddress,
-            recipient: followPolicy.recipient,
-          },
-          profileId: args.profile.id,
-          signless: false,
-          sponsored: args.sponsored ?? true,
-        };
-
-      case FollowPolicyType.ANYONE:
-        return {
-          kind: TransactionKind.FOLLOW_PROFILE,
-          profileId: args.profile.id,
-          signless: session.profile.signless,
-          sponsored: args.sponsored ?? true,
-        };
+/**
+ * @internal
+ */
+export function createFollowRequest(args: FollowArgs, session?: Session): FollowRequest {
+  invariant(
+    session?.authenticated,
+    'You must be authenticated to use this operation. Use `useLogin` hook to authenticate.',
+  );
+  invariant(
+    session.type === SessionType.WithProfile,
+    'You must have a profile to use this operation.',
+  );
 
-      case FollowPolicyType.UNKNOWN:
-        const result = await fetchModule({ implementation: followPolicy.contractAddress });
```

Comment:
no fetching module data for unknown module anymore?

Reply:
It is not needed when following, the information I was looking for (i.e. `sponsoredApproved` and `signlessApproved`) is repeated in the `UnknownFollowModuleSettings` for convenience.

On the other hand it's going to be needed when configuring a Profile's Follow Module and when creating a new publication.